pkg/retriever/parallelpeerretriever.go

```diff
@@ -182,8 +182,9 @@ func (retrieval *retrieval) RetrieveFromAsyncCandidates(asyncCandidates types.In
 
 	eventsCallback := func(evt types.RetrievalEvent) {
 		switch ret := evt.(type) {
-		case events.RetrievalEventFirstByte:
-			retrieval.Session.RecordFirstByteTime(evt.StorageProviderId(), ret.Duration())
+		case events.FirstByteEvent:
+			firstByteEvent := evt.(events.FirstByteEvent)
```

Comment:
shouldn't need this -- ret var already is FirstByteEvent.

Reply:
Ah, good eye. This one slipped by.